wp-job-manager-functions.php

```diff
@@ -303,66 +303,173 @@ function _wpjm_shuffle_featured_post_results_helper( $a, $b ) {
+			/**
+			 * Constructs SQL clauses that return posts which include or exclude the search term in the provided columns.
+			 *
+			 * @see WP_Query::parse_search()
+			 *
+			 * @param string $search_term     The search term to match.
+			 * @param bool   $is_excluding    Whether posts that match the search term should be excluded.
+			 * @param string $wildcard_search The wildcard character or empty string for exact matches.
+			 * @param array  $search_columns   The columns to check.
+			 *
+			 * @return array The SQL clauses.
+			 */
+			function job_manager_construct_post_conditions( $search_term, $is_excluding, $wildcard_search, $search_columns ) {
+				global $wpdb;
+
+				if ( $is_excluding ) {
+					$like_op = 'NOT LIKE';
+				} else {
+					$like_op = 'LIKE';
+				}
+
+				$like = $wildcard_search . $wpdb->esc_like( $search_term ) . $wildcard_search;
+
+				$conditions = [];
+				foreach ( $search_columns as $search_column ) {
+					//phpcs:disabled WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Variables are safe or escaped.
+					$conditions[] = $wpdb->prepare( "( {$wpdb->posts}.$search_column $like_op %s )", $like );
+				}
+
+				$allow_query_attachment_by_filename = apply_filters( 'wp_allow_query_attachment_by_filename', false );
+				if ( ! empty( $allow_query_attachment_by_filename ) ) {
+					$conditions[] = $wpdb->prepare( "(sq1.meta_value $like_op %s)", $like );
+					//phpcs:enabled WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				}
+
+				return $conditions;
+			}
+		}
 
 		/**
-		 * Filters the conditions to use when querying job listings. Resulting array is joined with OR statements.
-		 *
-		 * @since 1.26.0
-		 *
-		 * @param array  $conditions          Conditions to join by OR when querying job listings.
-		 * @param string $job_manager_keyword Search query.
+		 * This function aims to provide similar search functionality with WP core while also including meta and taxonomy terms
+		 * in the searched columns. The functionality of WP_Query::parse_search is replicated but with additional SQL
+		 * clauses which are generated in the job_manager_construct_secondary_conditions function.
 		 */
-		$conditions = apply_filters( 'job_listing_search_conditions', $conditions, $job_manager_keyword );
-		if ( empty( $conditions ) ) {
-			return $search;
+		$default_search_columns = [ 'post_title', 'post_excerpt', 'post_content' ];
+		$search_columns         = ! empty( $wp_query->query_vars['search_columns'] ) ? $wp_query->query_vars['search_columns'] : $default_search_columns;
+		if ( ! is_array( $search_columns ) ) {
+			$search_columns = [ $search_columns ];
 		}
 
-		$conditions_str = implode( ' OR ', $conditions );
+		$search_columns = (array) apply_filters( 'post_search_columns', $search_columns, $wp_query->query_vars['s'], $wp_query );
```

Comment:
It's coming from the core WP implementation. Generally I replicated what core does for searching as we now overwrite the whole search clause and not adding stuff on top.

This might be a bit risky as we might need to do more changes to replicate future core changes in WP_Query but since most of the stuff in WP_Query are public I think that they won't break backwards compatibility easily.

Reply:
Added a comment for the filter to specify that it comes from core WP